packages/main/src/methods/chat-session-helpers.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Content, POSSIBLE_ROLES, Part, Role } from "../../types";
+import { GoogleGenerativeAIError } from "../errors";
+
+const VALID_PART_FIELDS: Array<keyof Part> = [
```

Comment:
Maybe add a comment with a link to https://source.corp.google.com/piper///depot/google3/google/ai/generativelanguage/v1main/content.proto as a source of truth, unfortunately I don't think there's a public version?

Reply:
The are public docs on the rest API:
https://ai.google.dev/api/rest/v1beta/Content